Source/Magick.NET/Shared/Optimizers/ImageOptimizer.cs

```diff
@@ -252,7 +261,10 @@ private IImageOptimizer GetOptimizer(FileInfo file)
                     return optimizer;
             }
 
-            throw new MagickCorruptImageErrorException("Invalid format, supported formats are: " + SupportedFormats);
+            if (IgnoreUnsupportedFormats)
+                return null;
+            else
```

Comment:
I think I prefer this without the else, we don't need it here.

Reply:
Sure thing, elses now removed